modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+		add_action( 'wp_footer', array( $this, 'jetpack_social_menu_include_svg_icons' ), 9999 );
+	}
+
+	/**
+	 * Script & styles for admin widget form.
+	 */
+	public function enqueue_admin_scripts() {
+		wp_enqueue_script( 'jetpack-widget-social-icons-script', plugins_url( 'social-icons/social-icons-admin.js', __FILE__ ), array( 'jquery-ui-sortable' ), '20170506' );
+		wp_enqueue_style( 'jetpack-widget-social-icons-admin', plugins_url( 'social-icons/social-icons-admin.css', __FILE__ ), array(), '20170506' );
+	}
+
+	/**
+	 * Styles for front-end widget.
+	 */
+	public function enqueue_icon_scripts() {
+		wp_enqueue_style( 'jetpack-widget-social-icons-styles', plugins_url( 'social-icons/social-icons.css', __FILE__ ), array(), '20170506' );
+	}
+
+	/**
+	 * JavaScript for admin widget form.
+	 */
+	public function render_admin_js() {
+	?>
+		<script type="text/html" id="tmpl-jetpack-widget-social-icons-template">
+			<?php self::render_icons_template(); ?>
+		</script>
+	<?php
+	}
+
+	/**
+	 * Add SVG definitions to the footer.
+	 */
+	public function jetpack_social_menu_include_svg_icons() {
+		// Define SVG sprite file.
+		$svg_icons = dirname( dirname( __FILE__ ) ) . '/theme-tools/social-menu/social-menu.svg';
+
+		// If it exists, include it.
+		if ( file_exists( $svg_icons ) ) {
+			require_once( $svg_icons );
+		}
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @see WP_Widget::widget()
+	 *
+	 * @param array $args Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	public function widget( $args, $instance ) {
+		$instance = wp_parse_args( (array) $instance, $this->defaults );
+
+		/** This filter is documented in wp-includes/widgets/class-wp-widget-pages.php */
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'], $instance, $this->id_base );
+
+		echo $args['before_widget'];
+
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . esc_html( $title ) . $args['after_title'];
+		}
+
+		if ( ! empty( $instance['icons'] ) ) :
+
+			// Get supported social icons.
+			$social_labels = jetpack_social_menu_social_links_labels();
+			$social_icons  = jetpack_social_menu_social_links_icons();
+		?>
+
+			<ul class="jetpack-social-widget-list size-<?php echo esc_attr( $instance['icon-size'] ); ?>">
+
+				<?php foreach ( $instance['icons'] as $icon ) : ?>
+
+					<?php if ( ! empty( $icon['url'] ) ) : ?>
+						<li class="jetpack-social-widget-item">
+							<a href="<?php echo esc_url( $icon['url'], $this->protocols ); ?>">
+								<?php
+									// Add a label if there is supported URL.
+									foreach ( $social_labels as $attr => $value ) {
+										if ( false !== strpos( $icon['url'], $attr ) ) {
+											echo '<span class="screen-reader-text">' . esc_attr( $value ) . '</span>';
+										}
+									}
+
+									$icon_output = jetpack_social_menu_get_svg( array( 'icon' => 'chain' ) );
```

Comment:
Would it make more sense to cache this value so we're not calling `jetpack_social_menu_get_svg()` an extra time for every iteration of the foreach?